modules/dnn/test/test_layers.cpp

```diff
@@ -690,23 +697,94 @@ TEST_P(Scale_untrainable, Accuracy)
     float* refData = (float*)ref.data;
     float* weightsData = (float*)weights.data;
     int spatialSize = 1;
-    for (int i = axis + weightsDims; i < 4; ++i)
+    for (int i = axis + weightsDims - 1; i < 4; ++i)
         spatialSize *= inpShape[i];
-    for (int i = 0; i < ref.total(); ++i)
+    for (int num_samples = 0; num_samples < inpShape[0] ;num_samples++)
     {
-        float w = weightsData[(i / spatialSize) % weights.total()];
-        refData[i] = inpData[i] * w;
+        for (int i = 0; i < totalPerSample; ++i)
+        {
+            int index = i + num_samples * totalPerSample;
+            float w = weightsData[(i / spatialSize) % totalWeightsPerSample + num_samples * totalWeightsPerSample ];
+            refData[index] = inpData[index] * w;
+        }
     }
     normAssert(out, ref);
 }
 
 INSTANTIATE_TEST_CASE_P(Layer_Test, Scale_untrainable, Combine(
 /*input size*/   Values(Vec4i(2, 3, 4, 5)),
 /*axis, #dims*/  Values(Vec2i(0, 1), Vec2i(0, 2), Vec2i(0, 3), Vec2i(0, 4),
+                        Vec2i(1, 1), Vec2i(1, 2), Vec2i(1, 3), Vec2i(1, 4),
+                                     Vec2i(2, 1), Vec2i(2, 2), Vec2i(2, 3),
+                                                  Vec2i(3, 1), Vec2i(3, 2)),
```

Comment:
These tests are for scales that are computed at runtime. So their batchsize should be equal to Batchsize of Input0. So for all of them Scales are of size [N,C,H,W] / [N,C,H] / [N,H,W] / [N,C] / [N,H] / [N,W] / [N]

Extra added tests are 
Vec2i(1,4)
Vec2i(2,3) 
Vec2i(3,2)